lib/github_changelog_generator/argv_parser.rb

```diff
@@ -0,0 +1,224 @@
+          options[:since_commit] = v
+        end
+        opts.on("--max-issues NUMBER", Integer, "Maximum number of issues to fetch from GitHub. Default is unlimited.") do |max|
+          options[:max_issues] = max
+        end
+        opts.on("--release-url URL", "The URL to point to for release links, in printf format (with the tag as variable).") do |url|
+          options[:release_url] = url
+        end
+        opts.on("--github-site URL", "The Enterprise GitHub site where your project is hosted.") do |last|
+          options[:github_site] = last
+        end
+        opts.on("--github-api URL", "The enterprise endpoint to use for your GitHub API.") do |last|
+          options[:github_endpoint] = last
+        end
+        opts.on("--simple-list", "Create a simple list from issues and pull requests. Default is false.") do |v|
+          options[:simple_list] = v
+        end
+        opts.on("--future-release RELEASE-VERSION", "Put the unreleased changes in the specified release number.") do |future_release|
+          options[:future_release] = future_release
+        end
+        opts.on("--release-branch RELEASE-BRANCH", "Limit pull requests to the release branch, such as master or release.") do |release_branch|
+          options[:release_branch] = release_branch
+        end
+        opts.on("--[no-]http-cache", "Use HTTP Cache to cache GitHub API requests (useful for large repos). Default is true.") do |http_cache|
+          options[:http_cache] = http_cache
+        end
+        opts.on("--cache-file CACHE-FILE", "Filename to use for cache. Default is github-changelog-http-cache in a temporary directory.") do |cache_file|
+          options[:cache_file] = cache_file
+        end
+        opts.on("--cache-log CACHE-LOG", "Filename to use for cache log. Default is github-changelog-logger.log in a temporary directory.") do |cache_log|
+          options[:cache_log] = cache_log
+        end
+        opts.on("--config-file CONFIG-FILE", "Path to configuration file. Default is .github_changelog_generator.") do |config_file|
+          options[:config_file] = config_file
+        end
+        opts.on("--ssl-ca-file PATH", "Path to cacert.pem file. Default is a bundled lib/github_changelog_generator/ssl_certs/cacert.pem. Respects SSL_CA_PATH.") do |ssl_ca_file|
+          options[:ssl_ca_file] = ssl_ca_file
+        end
+        opts.on("--require x,y,z", Array, "Path to Ruby file(s) to require before generating changelog.") do |paths|
+          options[:require] = paths
+        end
+        opts.on("--[no-]verbose", "Run verbosely. Default is true.") do |v|
+          options[:verbose] = v
+        end
+        opts.on("-v", "--version", "Print version number.") do |_v|
+          puts "Version: #{GitHubChangelogGenerator::VERSION}"
+          exit
+        end
+        opts.on("-h", "--help", "Displays Help.") do
+          puts opts
+          exit
+        end
+      end
```

Comment:
When we have merged this, or earlier, we can figure out a way to not repeat (and need to maintain 2 copies of) these opts.on invocations.

Reply:
Not totally sure I follow. But where is the copy :)?

Reply:
Not totally sure I follow. But where is the copy?

Reply:
Ah, it was only me being momentarily confused, we're already at that place. Perfect!